docs/content/posts/2024-07-15-release-1.20.0.md

```diff
@@ -0,0 +1,560 @@
+
+-- leaves the number of buckets up to the connector.
+CREATE TABLE MyTable (uid BIGINT, name STRING) DISTRIBUTED BY (uid);
+
+-- only defines the number of buckets.
+CREATE TABLE MyTable (uid BIGINT, name STRING) DISTRIBUTED INTO 4 BUCKETS;
+```
+
+**More Information**
+* [FLINK-33494](https://issues.apache.org/jira/browse/FLINK-33494)
+* [FLIP-376](https://cwiki.apache.org/confluence/display/FLINK/FLIP-376%3A+Add+DISTRIBUTED+BY+clause)
+* [Documentation](https://nightlies.apache.org/flink/flink-docs-release-1.20/docs/dev/table/sql/create/#distributed)
+
+# State & Checkpoint Improvements
+
+## Unified File Merging Mechanism for Checkpoints
+
+The unified file merging mechanism for checkpointing is introduced to Flink 1.20 as an MVP ("minimum viable product") feature.
+It combines multiple small checkpoint files to into fewer larger files, which reduces the number of file creation
+and file deletion operations and alleviates the pressure of file system metadata management during checkpoints.
+
+The mechanism can be enabled by setting `execution.checkpointing.file-merging.enabled` to `true`. For more advanced options
+and principle behind this feature, please refer to the Checkpointing documentation.
+
+**More Information**
+* [FLINK-33494](https://issues.apache.org/jira/browse/FLINK-32070)
+* [FLIP-306](https://cwiki.apache.org/confluence/display/FLINK/FLIP-306%3A+Unified+File+Merging+Mechanism+for+Checkpoints)
+* [Documentation](https://nightlies.apache.org/flink/flink-docs-release-1.20/docs/dev/datastream/fault-tolerance/checkpointing/#unify-file-merging-mechanism-for-checkpoints-experimental)
+
+## Compaction of Small SST Files
+
+In some cases, the number of files produced by the RocksDB state backend grows indefinitely. In addition to the overhead caused by many small files,
+this behavior can cause the task state info to exceed the RPC message size limit and therefore lead to recovery or checkpoint failures.
+
+From release 1.20 on, Flink can merge such files in the background using the RocksDB API.
+
+**More Information**
+* [FLINK-26050](https://issues.apache.org/jira/browse/FLINK-26050)
+
+## Reorganizing all State & Checkpointing Configuration Options
+
+In Flink 1.20, all the options about state and checkpointing are reorganized and categorized by prefixes as listed below:
+1. `execution.checkpointing.*`: all configuration options associated with checkpointing and savepoints.
+2. `execution.state-recovery.*`: all configuration options related to state recovery.
+3. `state.*`: all configuration options related to the state accessing.
+   a. `state.backend.*`: configuration options for individual state backends, such as RocksDB.
+   b. `state.changelog.*`: configuration options for the state changelog, as outlined in FLIP-158, including the options for the "Durable Short-term Log" (DSTL).
+   c. `state.latency-track.*`: configuration options related to the latency tracking of state access.
```

Comment:
I think there are more configuration related changes. Maybe we should add a dedicated section for that, rather than separating configuration changes into sections of each component.